app/controllers/companies_controller.rb

```diff
@@ -10,13 +10,15 @@ def show
   def create
     company = Company.new(company_params)
     if company.save
-      User.create(email: company_params[:email],
-                  name: company_params[:name],
-                  title: company_params[:title],
-                  password: User.generate_password,
-                  company_id: company.id
-                 )
+      password = User.generate_password
+      user = User.create(email: company_params[:email],
+                          name: company_params[:name],
+                          title: company_params[:title],
+                          password: password,
+                          company_id: company.id
+                          )
       flash[:success] = "Welcome #{company.organization}"
+      UserEmailer.send_welcome_email(user, password).deliver_now
```

Comment:
I really like the logic here! Pretty cool making both a company and a user. If we have time later, we may want to refactor some of this logic out of the controller. 

Here is the lesson Josh gave us on refactoring mailers with callbacks. 
<a href=https://www.dropbox.com/sh/htzj11lf6nlynmh/AAD_MVu_ZfvtuISu2MndVGsca/R3%20Professional%20Rails%20Applications/Advanced%20ActiveRecord%20-%20Callbacks%20and%20Transactions.key?dl=0> Mailers and callbacks lesson </a>